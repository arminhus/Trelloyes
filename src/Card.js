import React from "react";
import "./Card.css";

let Card = props => {
  return (
    <div className="Card">
      <button type="button">delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
      {/* <h3>{state.title}</h3> */}
    </div>
  );
};
export default Card;
