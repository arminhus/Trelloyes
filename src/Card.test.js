import React from "react";
import ReactDom from "react-dom";
import Card from "./Card";
import renderer, { create } from "react-test-renderer";

describe("Card component test", () => {
  test("should render with crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Card />, div);
    ReactDom.unmountComponentAtNode(div);
  });

  test("should render card first card in first list", () => {
    const tree = renderer
      .create(<Card title="first card" content="lorem ipsum" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //   snapshot tests
  test("should match the snapshot for Card", () => {
    const card = create(<Card />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});
