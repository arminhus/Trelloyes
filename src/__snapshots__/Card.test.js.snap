// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test should match the snapshot for Card 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`Card component test should render card first card in first list 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    first card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
